From efa6fc548eafdeaf103e559fb07b1a82526dd0d4 Mon Sep 17 00:00:00 2001
From: Santosh Shukla <sshukla@marvell.com>
Date: Mon, 3 Sep 2018 20:54:04 +0530
Subject: [PATCH 0186/1051] gpio: thunderx: Fixes
 config_mrvl_octeontx2_el0_intr
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If disabled `CONFIG_MRVL_OCTEONTX_EL0_INTR` then below build fix:

  CC      drivers/firmware/efi/arm-runtime.o
drivers/gpio/gpio-thunderx.c: In function ‘thunderx_gpio_remove’:
drivers/gpio/gpio-thunderx.c:901:17: error: ‘otx_class’ undeclared (first use in this function); did you mean ‘class’?
  device_destroy(otx_class, otx_dev);
                 ^~~~~~~~~
                 class
drivers/gpio/gpio-thunderx.c:901:17: note: each undeclared identifier is reported only once for each function it appears in
drivers/gpio/gpio-thunderx.c:901:28: error: ‘otx_dev’ undeclared (first use in this function); did you mean ‘pci_dev’?
  device_destroy(otx_class, otx_dev);
                            ^~~~~~~
                            pci_dev
drivers/gpio/gpio-thunderx.c:903:2: error: implicit declaration of function ‘cdev_del’; did you mean ‘dev_dbg’? [-Werror=implicit-function-declaration]
  cdev_del(otx_cdev);
  ^~~~~~~~
  dev_dbg
drivers/gpio/gpio-thunderx.c:903:11: error: ‘otx_cdev’ undeclared (first use in this function); did you mean ‘otx_dev’?
  cdev_del(otx_cdev);
           ^~~~~~~~
           otx_dev
  CC      drivers/gpio/gpio-xgene.o
drivers/gpio/gpio-thunderx.c:906:2: error: implicit declaration of function ‘task_cleanup_handler_remove’; did you mean ‘wakeup_source_remove’? [-Werror=implicit-function-declaration]
  task_cleanup_handler_remove(cleanup_el3_irqs);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  wakeup_source_remove
drivers/gpio/gpio-thunderx.c:906:30: error: ‘cleanup_el3_irqs’ undeclared (first use in this function); did you mean ‘cleanup_module’?
  task_cleanup_handler_remove(cleanup_el3_irqs);
                              ^~~~~~~~~~~~~~~~
                              cleanup_module
cc1: some warnings being treated as errors
scripts/Makefile.build:328: recipe for target 'drivers/gpio/gpio-thunderx.o' failed

Signed-off-by: Santosh Shukla <sshukla@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/gpio/gpio-thunderx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index 2f88eea46b43..c6f81abc88d9 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -886,12 +886,14 @@ static void thunderx_gpio_remove(struct pci_dev *pdev)
 
 	pci_set_drvdata(pdev, NULL);
 
+#ifdef CONFIG_MRVL_OCTEONTX_EL0_INTR
 	device_destroy(otx_class, otx_dev);
 	class_destroy(otx_class);
 	cdev_del(otx_cdev);
 	unregister_chrdev_region(otx_dev, 1);
 
 	task_cleanup_handler_remove(cleanup_el3_irqs);
+#endif
 }
 
 static const struct pci_device_id thunderx_gpio_id_table[] = {
-- 
2.17.1

